### PR TITLE
Server-side gift card catalog management

### PR DESCRIPTION
**Pull request by msalcala11**
1. Moves offered gift cards and assets to the server
2. Caches gift card images locally on the device on 1st load for fast rendering and for the ability to view previously purchased gift cards properly in offline mode

This will allow adding new gift cards to the catalog without having to release a new version of the app.

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export function promiseSerial(tasks: Array<() => Promise<any>>) {
```
Comment (cmgustavo) [inline src/utils.ts]: What 's the purpose of this file?

Comment (msalcala11) [reply src/utils.ts]: This is where commonly-used, generic functions will go.

Comment (msalcala11) [reply src/utils.ts]: The first function I've added (`promiseSerial`) executes a lists of async tasks in series.

Comment (cmgustavo) [reply src/utils.ts]: ACK!

**Comment by gabrielbazan7** [comment]
CODE ACK

**Pull request merged**